LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationConfigurationService.cs

```diff
@@ -60,20 +62,40 @@ private async Task<TwinCollection> GetTwinDesiredPropertiesAsync(StationEui stat
 
             try
             {
-                return await this.cache.GetOrCreateAsync(cacheKey, async cacheEntry =>
+                LoRaProcessingException exception = null;
+                var cacheValue = await this.cache.GetOrCreateAsync(cacheKey, async cacheEntry =>
                 {
-                    _ = cacheEntry.SetAbsoluteExpiration(CacheTimeout);
-                    var key = await this.loRaDeviceApiService.GetPrimaryKeyByEuiAsync(stationEui);
-                    if (string.IsNullOrEmpty(key))
+                    try
                     {
-                        throw new LoRaProcessingException($"The configuration request of station '{stationEui}' did not match any configuration in IoT Hub. If you expect this connection request to succeed, make sure to provision the Basics Station in the device registry.",
-                                                          LoRaProcessingErrorCode.InvalidDeviceConfiguration);
+                        _ = cacheEntry.SetAbsoluteExpiration(CacheTimeout);
+
+                        var key = await this.loRaDeviceApiService.GetPrimaryKeyByEuiAsync(stationEui);
+                        if (string.IsNullOrEmpty(key))
+                        {
+                            throw new LoRaProcessingException($"The configuration request of station '{stationEui}' did not match any configuration in IoT Hub. If you expect this connection request to succeed, make sure to provision the Basics Station in the device registry.",
+                                                                LoRaProcessingErrorCode.InvalidDeviceConfiguration);
+                        }
+
+                        using var client = this.loRaDeviceFactory.CreateDeviceClient(stationEui.ToString(), key);
+                        var twin = await client.GetTwinAsync(cancellationToken);
+                        return twin.Properties.Desired;
                     }
+                    catch (LoRaProcessingException ex)
+                    {
+                        exception = ex; // keeps the exception to re-throw after creating the cache entry
```

Comment:
The better way to capture exceptions for re-throw from _elsewhere_ is to use `ExceptionDispatchInfo` to capture the exception and have it re-throw, which has the effect of preserving the stack trace and more accurately representing the original throw-site.

I would recommend updating as follows:

```diff
diff --git a/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationConfigurationService.cs b/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationConfigurationService.cs
index d15901f6..63413458 100644
--- a/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationConfigurationService.cs
+++ b/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationConfigurationService.cs
@@ -4,6 +4,7 @@
 namespace LoRaWan.NetworkServer.BasicsStation
 {
     using System;
+    using System.Runtime.ExceptionServices;
     using System.Text.Json;
     using System.Threading;
     using System.Threading.Tasks;
@@ -62,7 +63,7 @@ namespace LoRaWan.NetworkServer.BasicsStation
 
             try
             {
-                LoRaProcessingException exception = null;
+                ExceptionDispatchInfo edi = null;
                 var cacheValue = await this.cache.GetOrCreateAsync(cacheKey, async cacheEntry =>
                 {
                     try
@@ -82,18 +83,17 @@ namespace LoRaWan.NetworkServer.BasicsStation
                     }
                     catch (LoRaProcessingException ex)
                     {
-                        exception = ex; // keeps the exception to re-throw after creating the cache entry
+                        edi = ExceptionDispatchInfo.Capture(ex); // keeps the exception to re-throw after creating the cache entry
 
                         _ = cacheEntry.SetAbsoluteExpiration(CacheLbsRateLimitingTimeout);
-                        return new TwinCollection(); // dummy cache entries are used in order to "rate limit" LBS so that we don't contact IoTHub too often 
+                        return new TwinCollection(); // dummy cache entries are used in order to "rate limit" LBS so that we don't contact IoTHub too often
                     }
                 });
 
 #pragma warning disable CA1508 // Avoid dead conditional code
                 // false positive, variable is conditionally not null when an exception is thrown
-                if (exception != null)
+                edi?.Throw();
 #pragma warning restore CA1508 // Avoid dead conditional code
-                    throw exception;
 
                 return cacheValue;
             }
```

You can copy [the above diff](https://github.com/Azure/iotedge-lorawan-starterkit/files/7932150/diff.txt) to the clipboard and apply it from PowerShell using: `Get-Clipboard | git apply`

See the [**Remarks** section of `ExceptionDispatchInfo`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo?view=net-6.0#remarks) for more background:

> The [`ExceptionDispatchInfo`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo?view=net-6.0) object stores the stack trace information and Watson information that the exception contains at the point where it is captured. The exception can be thrown at another time and possibly on another thread by calling the [`ExceptionDispatchInfo.Throw`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.throw?view=net-6.0) method. The exception is thrown as if it had flowed from the point where it was captured to the point where the [`Throw`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.throw?view=net-6.0) method is called.